### PR TITLE
Add debugger tests

### PR DESCRIPTION
**Pull request by mfilippov**
1. Fix Attach test.
2. Add test on incorrect events behaviour when vm started with suspend

### CODE REVIEW

File: mcs/class/Mono.Debugger.Soft/Test/dtest.cs [hunk 1]
```diff
@@ -4380,20 +4381,32 @@ Event step_out_await (string method, Event e)
 		assert_location(e, "ss_nested_arg");
 	}
 
+	private static int GetFreePort ()
+	{
```
Comment (vargaz) [inline mcs/class/Mono.Debugger.Soft/Test/dtest.cs]: Methods are private by default, no need to state that.

File: mcs/class/Mono.Debugger.Soft/Test/dtest.cs [hunk 2]
```diff
@@ -4380,20 +4381,32 @@ Event step_out_await (string method, Event e)
...
+		var s = new Socket (AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+		s.Bind (new IPEndPoint(IPAddress.Loopback, 0));
+		var port = ((IPEndPoint)s.LocalEndPoint).Port;
+		s.Close ();
+		return port;
+	}
+
+	private static VirtualMachine ConnectToPort (int port) {
+		var ep = new IPEndPoint (IPAddress.Loopback, port);
```
Comment (vargaz) [inline mcs/class/Mono.Debugger.Soft/Test/dtest.cs]: Ditto.

File: mcs/class/Mono.Debugger.Soft/Test/dtest.cs [hunk 3]
```diff
@@ -4526,5 +4526,122 @@ public void Bug59649 ()
 		AssertValue (3.0, f);
 
 	}
+
+	private void HandleEvents()
+	{
```
Comment (vargaz) [inline mcs/class/Mono.Debugger.Soft/Test/dtest.cs]: Ditto.

File: mcs/class/Mono.Debugger.Soft/Test/dtest.cs [hunk 4]
```diff
@@ -4526,5 +4526,122 @@ public void Bug59649 ()
...
+				}
+			}
+		}
+
+		if (failMessage != null)
+			Assert.Fail(failMessage);
+	}
+
+	private void ShouldNotSendUnexpectedTypeLoadEventsAndInvalidSuspendPolicyAfterAttach (int port)
+	{
```
Comment (vargaz) [inline mcs/class/Mono.Debugger.Soft/Test/dtest.cs]: Ditto.

File: mcs/class/Mono.Debugger.Soft/Test/dtest.cs [hunk 5]
```diff
@@ -4526,5 +4526,122 @@ public void Bug59649 ()
...
+		var process = Diag.Process.Start (pi);
+
+		ShouldNotSendUnexpectedTypeLoadEventsAndInvalidSuspendPolicyAfterAttach (port);
+	}
+
+	[Test]
+	[Category("NotWorking")]
+	public void ShouldNotSendUnexpectedTypeLoadEventsAndInvalidSuspendPolicyAfterAttachIfSuspendIsFalse()
+	{
+		vm.Exit (0);
```
Comment (vargaz) [inline mcs/class/Mono.Debugger.Soft/Test/dtest.cs]: The idea is that the debuggee collects the events and sends them when the debugger attaches.

Comment (mfilippov) [reply mcs/class/Mono.Debugger.Soft/Test/dtest.cs]: When we starting HandleEvents in test debugger already attached and stopped at the breakpoint. As I understand after that we shouldn't receive "TypeLoad" or "Breakpoint" event in this time. If we received event after the stop at the breakpoint and call Resume after TypeLoad event we incorrect behaviour.

**Comment by luhenry** [comment]
@vargaz could you please re-review?

**Comment by mfilippov** [comment]
@vargaz ping

**Comment by mfilippov** [comment]
@luhenry @vargaz Any news?

**Comment by thaystg** [comment]
@mfilippov, for me it looks ok, could you please resolve the conflicts and then I can approve?

**Comment by mfilippov** [comment]
@thaystg Thank you for the review.

**Comment by mfilippov** [comment]
@thaystg I fixed merge conflict and tests.

**Comment by luhenry** [comment]
@monojenkins build failed

**Comment by mfilippov** [comment]
@luhenry I added API in VirtualMachine with the timeout to avoid an infinite wait.

**Comment by mfilippov** [comment]
@thaystg ready to review

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by luhenry** [comment]
@akoeplinger we need your magic touch here to fix the API diff :)

**Comment by akoeplinger** [comment]
@monojenkins commit apidiff

**Pull request merged**